tests/framework/e2e/etcd_process.go

```diff
@@ -183,6 +184,17 @@ func (ep *EtcdServerProcess) Logs() LogsExpect {
 	return ep.proc
 }
 
+func (cfg *EtcdServerProcessConfig) SetInitialCluster(nodes []string, initialClusterState string) {
+	cfg.InitialCluster = strings.Join(nodes, ",")
+	cfg.Args = append(cfg.Args, "--initial-cluster", cfg.InitialCluster)
+	cfg.Args = append(cfg.Args, "--initial-cluster-state", initialClusterState)
+}
+
+func (cfg *EtcdServerProcessConfig) EnableDiscovery(token string, endpoints []string) {
+	cfg.Args = append(cfg.Args, fmt.Sprintf("--discovery-token=%s", token))
+	cfg.Args = append(cfg.Args, fmt.Sprintf("--discovery-endpoints=%s", strings.Join(endpoints, ",")))
+}
```

Comment:
let's not implement setters for configuration structs. I think we should avoid mutating args when constructing EtcdServerProcess. One solution would be to separate args provided by caller (`extraArgs`) from args generated from configuration fields (`configArgs`).

Reply:
@biosvs  After you resolve https://github.com/etcd-io/etcd/pull/14404#discussion_r969102907, then this comment should can be resolved automatically.